bluebrain/repo-bluebrain/packages/neurodamus/package.py

```diff
@@ -0,0 +1,372 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+import shutil
+from contextlib import contextmanager
+
+from llnl.util import tty
+
+from spack.package import *
+
+# Definitions
+_CORENRN_MODLIST_FNAME = "coreneuron_modlist.txt"
+_BUILD_NEURODAMUS_FNAME = "build_neurodamus.sh"
+
+
+class Neurodamus(Package):
+    """The next-generation AllInOne Neurodamus deployment.
+
+    Neurodamus includes BBP's simulation suite with all internal sim models.
+    """
+
+    homepage = "https://bbpgitlab.epfl.ch/hpc/sim/neurodamus-models"
+    git = "ssh://git@bbpgitlab.epfl.ch/hpc/sim/neurodamus-models.git"
+
+    version("develop", submodules=True)
+    # TODO: Shall we let the version scheme be different to avoid mixing with old neurodamus-**?
+    version("2023.05", tag="0.0.2", submodules=True)
+    version("2023.04", tag="0.0.2", submodules=True)
+
+    variant("synapsetool", default=True, description="Enable SynapseTool reader (for edges)")
+    variant("mvdtool", default=True, description="Enable MVDTool reader (for nodes)")
+    variant("coreneuron", default=False, description="Enable CoreNEURON Support")
+    variant("caliper", default=False, description="Enable Caliper instrumentation")
+
+    # neuron/corenrn get linked automatically when using nrnivmodl[-core]
+    # Dont duplicate the link dependency (only 'build' and 'run')
+    depends_on("neuron+mpi", type=("build", "run"))
+    depends_on("neuron+caliper", when="+caliper", type=("build", "run"))
+    depends_on("gettext", when="^neuron+binary")
+
+    depends_on("py-neurodamus", type=("build", "run"))
+    # Note: We dont request link to MPI so that mpicc can do what is best
+    # and dont rpath it so we stay dynamic.
+    # 'run' mode will load the same mpi module
+    depends_on("mpi", type=("build", "run"))
+    depends_on("hdf5+mpi")
+    depends_on("reportinglib")
+    depends_on("libsonata-report")
+    depends_on("synapsetool+mpi", when="+synapsetool")
+    depends_on("py-mvdtool+mpi", type="run", when="+mvdtool")
```

Comment:
So... we don't "link" to MPI ourselves, but our link dependencies may? Or are all these MPI dependencies vetted to not RPATH MPI?